### ISSUE TITLE
Redraw the menus after a reordering

### ISSUE THREAD

**Issue by afrancoto** [afrancoto/WebMenu#14]
The `change_order()` function in WebMenu_config.py works greatly: the settings are written in the right new order in dconf. Here it is:

<pre>
    def change_order(self, widget, treeview, liststore, direction):
    #TODO: Need to update the menus correctly
    global services, services_order
    (model, tree_iter) =  treeview.get_selection().get_selected()
        service = model.get_value(tree_iter,0)

    moved_one_index=services_order.index(service)
    if direction is 'down': moved_two_index=moved_one_index + 1
    if direction is 'up': moved_two_index=moved_one_index - 1
    
    services_order[moved_one_index]=services_order[moved_two_index]
    services_order[moved_two_index]=service
    self.settings['services-order']=services_order

    liststore.clear()
    for service in services_order: liststore.append([service, services[service][1], services[service][2], services[service][3], services[service][4]])
    treeview.set_cursor(moved_two_index)
    return
</pre>

When it is called the `service-order` key is changed, so the `apply_settings()`  function in WebMenu.py is called with the parameter `key='changed'`:

<pre>
  def apply_settings(self, settings, key, shell, config):
    global services, services_order, other_settings

    #The global variables are updated if changed
    if (key is not 'all') and (key is not 'firsttime'): 
    services = self.settings['services']
    services_order = self.settings['services-order']
        other_settings = self.settings['other-settings']
</pre>

The global variables are updated

<pre>
    config.check_services_order()
    
    self.draw_menu(shell)#Redraws the menus
    self.draw_context_menu(shell)
</pre>

`check_services_order` is a function in WebMenu_config.py that fixes wrong settings... if everything is okay it does nothing.
The draw functions are called! The rest of the `apply_settings()` hides unwanted options and has nothing to do with their order.

<pre>
  def draw_menu(self, shell):
    global action_group
    global ui_id

    ui_manager = shell.props.ui_manager
    menu_not_drawn=False #Usually the function is called by apply_settings, so the menu is already drawn

    try: ui_id
    except NameError: menu_not_drawn=True #If the ui_context_id does not exist, the menu is not yet drawn, so it is not deleted

    if not menu_not_drawn :
    ui_manager.remove_ui(ui_id) #Delete a previous drawn menu
    ui_manager.remove_action_group(action_group)
    del ui_id, action_group
</pre>

if the menus have been drawn by a previous `draw_menu()` call, they are deleted to make place for the new ones. 

<pre>
  [...]
    #0.2.X Album SubMenu Items
    ui_album=''
    for service in services_order:
    if services[service][1] is not '':  #If the album URL is empty does not display the option
        #Add a new submenu item 
            action_name = 'album_%s' % service
            ui_album += '<menuitem name="AL_%s" action="%s"/>' % (service, action_name) #ui_album is the Album submenu
            #Create the action
            action = Gtk.Action( action_name, service, _('Look for the current album on %s' % service), '' )
            action.connect( 'activate', self.unique_search_function, shell, 1, service)
            action_group.add_action( action )
</pre>

The menu items and actions are written...

<pre>
  [...]
    ui = web_menu_item % (ui_album, ui_artist) #Adds ui_album and ui_artist to the webmenu
    os.system('echo "'+ui+'"')
</pre>

..and added to the menu. With the echo command i could see that the `ui` variable is written 
##### WITH THE RIGHT NEW ORDER!!

<pre>
    ui_manager.insert_action_group(action_group)
    ui_id = ui_manager.add_ui_from_string(ui)
    ui_manager.ensure_update()
</pre>

 the `ui_manager` adds the menu but... it is in the wrong order (sometimes the old one, sometimes an item goes under the separator over the 'All' item, it's a mess) :(

Do you have any idea?

Sorry for the long long long post :P

**Comment by asermax** [comment]
I can't find where this bug comes from, but I realized that, at least in my tests, the plugin starts misbehaving when adding a new service and THEN changing it's order. 
Thinking that maybe it haves to do with remnants of the previous drawn menu still interfering with the new disposition of the menu, I added 'ui_manager.ensure_update()' after the deletion of the old ui. That seemed to work (again, given my tests, I might be getting lucky xD), so please test it out and let me know what your results are :F

**Comment by afrancoto** [comment]
I added the `ui_manager.ensure_update` but nothing changed :/
So, I changed the update policy of the settings: instead of being updated in real time, the new settings are kept in global variables and written in dconf only when the Apply button is pressed. Now it works! :D 
Maybe I just didn't give enough time to the ui_manager, even if it seems strange to me...

I also corrected a mistake i made in the function `website_toggled_from_list`: I was updating the service of the selected row instead of the one the checkbox belongs to :P

Uh, now when i add a service and, after that, I click on 'Apply', the manage window doesn't close! Even if the apply_settings function is:

<pre>
    def apply_settings(self, widget, data=None):
    self.settings['services']=services
    self.settings['services-order']=services_order
    self.settings['other-settings']=other_settings
    self.window.destroy()
</pre>

Mah!
